kedro-datasets/kedro_datasets/polars/polars_dataset.py

```diff
@@ -0,0 +1,272 @@
+                " https://pola-rs.github.io/polars/py-polars/html/reference/io.html"
+            )
+
+        _fs_args = deepcopy(fs_args) or {}
+        _credentials = deepcopy(credentials) or {}
+
+        protocol, path = get_protocol_and_path(filepath, version)
+        if protocol == "file":
+            _fs_args.setdefault("auto_mkdir", True)
+
+        self._protocol = protocol
+        self._storage_options = {**_credentials, **_fs_args}
+        self._fs = fsspec.filesystem(self._protocol, **self._storage_options)
+
+        self.metadata = metadata
+
+        super().__init__(
+            filepath=PurePosixPath(path),
+            version=version,
+            exists_function=self._fs.exists,
+            glob_function=self._fs.glob,
+        )
+
+        # Handle default load and save arguments
+        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        if "storage_options" in self._save_args or "storage_options" in self._load_args:
+            logger.warning(
+                "Dropping 'storage_options' for %s, "
+                "please specify them under 'fs_args' or 'credentials'.",
+                self._filepath,
+            )
+            self._save_args.pop("storage_options", None)
+            self._load_args.pop("storage_options", None)
+
+    def _describe(self) -> Dict[str, Any]:
+        return {
+            "filepath": self._filepath,
+            "protocol": self._protocol,
+            "load_args": self._load_args,
+            "save_args": self._save_args,
+            "version": self._version,
+        }
+
+    def _load(self) -> pl.LazyFrame:
+        load_path = str(self._get_load_path())
+
+        if self._protocol == "file":
+            # With local filesystems, we can use Polar's build-in I/O method:
+            load_method = getattr(pl, f"scan_{self._file_format}", None)
+            return load_method(load_path, **self._load_args)
+
+        # For object storage, we use pyarrow for I/O:
+        fsspec.filesystem("s3")
+        dataset = ds.dataset(load_path, filesystem=self._fs, format=self._file_format)
+        return pl.scan_pyarrow_dataset(dataset)
+
+    def _save(self, data: Union[pl.DataFrame, pl.LazyFrame]) -> None:
+        save_path = get_filepath_str(self._get_save_path(), self._protocol)
+        if Path(save_path).is_dir():
+            raise DatasetError(
+                f"Saving {self.__class__.__name__} to a directory is not supported."
+            )
+
+        if "partition_cols" in self._save_args:
+            raise DatasetError(
+                f"{self.__class__.__name__} does not support save argument "
+                f"'partition_cols'. Please use 'kedro.io.PartitionedDataset' instead."
+            )
+
+        BytesIO()
```

Comment:
What does this do?

Reply:
Nothing, that's probably a left-over of something (or a copy-paste error)